src/plan/global.rs

```diff
@@ -437,7 +432,11 @@ impl<VM: VMBinding> BasePlan<VM> {
         constraints: &'static PlanConstraints,
     ) -> BasePlan<VM> {
         let stats = Stats::new();
-        let ctr = stats.new_event_counter("gc.num", true, true);
+        // Initializing the analysis manager and routines
+        #[cfg(feature = "analysis")]
+        let mut analysis_manager = Mutex::new(AnalysisManager::new());
+        #[cfg(feature = "analysis")]
+        analysis_manager.lock().unwrap().initialize_routines(&stats);
```

Comment:
Is there any reason we need to `new()` then `initialize_routines()`? You could make `initialize_routines()` a private function, and call it in `new()`.